### PR TITLE
NEW 88190: update submission summary screen design

### PR DESCRIPTION
**Pull request by jessicamoorthy**

### CODE REVIEW

File: src/js/utils/SiteConfigParser.js [hunk 1]
```diff
@@ -20,7 +20,8 @@ export default class SiteConfigParser {
       logoPath: toString(get(jsonObject, "LogoPath", '') || DefaultLogoImage),
       homePageBackgroundImagePath: toString(get(jsonObject, "HomePageBackgroundImagePath", '') || DefaultBackgroundImage),
       pdfHeaderImageLink: toString(get(jsonObject, "PdfHeaderImageLink", '') || DefaultPDFHeaderImage),
-      pdfFooterImageLink: toString(get(jsonObject, "PdfFooterImageLink", '') || DefaultPDFFooterImage)
+      pdfFooterImageLink: toString(get(jsonObject, "PdfFooterImageLink", '') || DefaultPDFFooterImage),
+      securityTeamEmail: toString(get(jsonObject, "SecurityTeamEmail") || 'security.architects@nzta.govt.nz'),
```
Comment (roopamjain01) [inline src/js/utils/SiteConfigParser.js:24]: we use same code for other agency as well.

File: src/js/services/QuestionnaireDataService.js [hunk 2]
```diff
@@ -233,12 +241,20 @@ query {
...
             };
             return taskSubmission;
           }),
         collaborators: UserParser.parserCollaboratorsFromJSON(_.get(submissionJSON, "CollaboratorList", [])),
-        riskResults: _.has(submissionJSON, 'RiskResultData') ? JSON.parse(_.get(submissionJSON, "RiskResultData", "[]")) : "[]"
-      },
+        riskResults: _.has(submissionJSON, 'RiskResultData') ? JSON.parse(_.get(submissionJSON, "RiskResultData", "[]")) : "[]",
+        created: _.toString(_.get(submissionJSON, "Created", "")),
+        releaseDate: _.toString(_.get(submissionJSON, "ReleaseDate", "")),
+        siteConfig: SiteConfigParser.parseSiteConfigFromJSON(siteData),
```
Comment (roopamjain01) [inline src/js/services/QuestionnaireDataService.js:255]: siteConfig are mentioned 2 time (line number 185/184 and 255)

**Comment by elliot-sawyer** [comment]
I'm not able to resolve the merge conflict here without compiling with the correct version of node/npm. Best to avoid unintentional breakages, so will leave to @jessicamoorthy or @roopamjain01

File: src/js/components/Questionnaire/Summary.js [hunk 3]
```diff
@@ -319,30 +328,95 @@ class Summary extends Component<Props> {
...
+              {moment(submission.created).format("DD/MM/YYYY")}
+            </div>
+            <div class="content-container">
+              <div className="content">Go live date: </div>
+              {submission.releaseDate ? moment(submission.releaseDate).format("DD/MM/YYYY") : ''}
+            </div>
+          </div>
+          <div class="column">
+            <div class="content-container">
+              <div className="content">Submitted by:</div>{submitter.name}
```
Comment (roopamjain01) [inline src/js/components/Questionnaire/Summary.js:383]: Why are you using "div" tag for the inline element as well?
Can we plese use span for that?
Please read all the html tag and how and where to use that.
And also for "submitted by:" you are using the div tag but not for the "submitter.name", both should inside the span container.

I can see the same issue in the entire file.

File: src/js/components/Questionnaire/Summary.js [hunk 4]
```diff
@@ -319,30 +328,95 @@ class Summary extends Component<Props> {
...
-        <div><b>Status:</b> {prettifyStatus(submission.status)}</div>
+      <div class="submission-details-container">
+        <h3>Submission details</h3>
+        <div class="row">
+          <div class="column">
+            <div className="product-name">{submission.productName}</div>
+            <div>{submission.questionnaireTitle}</div>
+          </div>
+          <div>
+            <span className="submission-status"> <img src={statusIcon} /> {status} </span>
```
Comment (roopamjain01) [inline src/js/components/Questionnaire/Summary.js:367]: Please remove unwanted space before  tag and after {status}.

File: src/js/components/Questionnaire/Summary.js [hunk 5]
```diff
@@ -319,30 +328,95 @@ class Summary extends Component<Props> {
...
+          </div>
+        </div>
+        <div class="row-divider">
+          <div class="column">
+            <div class="content-container">
+              <div className="content">Submission created: </div>
```
Comment (roopamjain01) [inline src/js/components/Questionnaire/Summary.js:373]: Do we really need div tag here?

File: src/js/components/Questionnaire/Summary.js [hunk 6]
```diff
@@ -319,30 +328,95 @@ class Summary extends Component<Props> {
...
+              {moment(submission.created).format("DD/MM/YYYY")}
+            </div>
+            <div class="content-container">
+              <div className="content">Go live date: </div>
```
Comment (roopamjain01) [inline src/js/components/Questionnaire/Summary.js:377]: Same here, please use span. We don't need to use <div> for everything for allthe time. There are other tag also available in the html to solve the purpose.

File: src/js/components/Questionnaire/Summary.js [hunk 7]
```diff
@@ -294,23 +301,25 @@ class Summary extends Component<Props> {
 
   renderCollboratorsInfo(submission: Submission, members) {
     return (
-      <div>
+      <div className="collaborators-container">
         <h4>Collaborators</h4>
-        <div><b>Add people to help complete your submission.</b></div>
+        <div className="collaborator-message">
+          You can add people to help complete your submission. Please contact the
+          <a href = {"mailto:" + this.props.securityTeamEmail}> security team</a> for more information.
```
Comment (roopamjain01) [inline src/js/components/Questionnaire/Summary.js:308]: Underline should start from the "s" only. and Same issue we have with the back button link.
Right now it is start from the space in the left site.

File: src/scss/components/Common/_Table.scss [hunk 8]
```diff
@@ -10,9 +10,36 @@ $white-background: rgba(255, 255, 255, 0.7);
...
 
   td {
     font-size: 11px;
   }
+
+  .table th {
+    border-top: none;
+  }
+
+  .task-thead, .submission-approval-thead {
```
Comment (roopamjain01) [inline src/scss/components/Common/_Table.scss:24]: Please move the below code in the relevent files.

File: src/js/utils/SiteConfigParser.js [hunk 9]
```diff
@@ -20,7 +20,8 @@ export default class SiteConfigParser {
       logoPath: toString(get(jsonObject, "LogoPath", '') || DefaultLogoImage),
       homePageBackgroundImagePath: toString(get(jsonObject, "HomePageBackgroundImagePath", '') || DefaultBackgroundImage),
       pdfHeaderImageLink: toString(get(jsonObject, "PdfHeaderImageLink", '') || DefaultPDFHeaderImage),
-      pdfFooterImageLink: toString(get(jsonObject, "PdfFooterImageLink", '') || DefaultPDFFooterImage)
+      pdfFooterImageLink: toString(get(jsonObject, "PdfFooterImageLink", '') || DefaultPDFFooterImage),
+      securityTeamEmail: toString(get(jsonObject, "SecurityTeamEmail") || 'securityteamemail@email.com'),
```
Comment (roopamjain01) [inline src/js/utils/SiteConfigParser.js:24]: Can it be leave empty, does anyone ask you to use this email by default "securityteamemail@email.com"?

File: src/img/icons/add.svg [hunk 10]
```diff
@@ -1,3 +1,3 @@
-<svg class="svg-icon" viewBox="0 0 20 20">
-							<path d="M14.613,10c0,0.23-0.188,0.419-0.419,0.419H10.42v3.774c0,0.23-0.189,0.42-0.42,0.42s-0.419-0.189-0.419-0.42v-3.774H5.806c-0.23,0-0.419-0.189-0.419-0.419s0.189-0.419,0.419-0.419h3.775V5.806c0-0.23,0.189-0.419,0.419-0.419s0.42,0.189,0.42,0.419v3.775h3.774C14.425,9.581,14.613,9.77,14.613,10 M17.969,10c0,4.401-3.567,7.969-7.969,7.969c-4.402,0-7.969-3.567-7.969-7.969c0-4.402,3.567-7.969,7.969-7.969C14.401,2.031,17.969,5.598,17.969,10 M17.13,10c0-3.932-3.198-7.13-7.13-7.13S2.87,6.068,2.87,10c0,3.933,3.198,7.13,7.13,7.13S17.13,13.933,17.13,10"></path>
-						</svg>
+<svg class="svg-icon" viewBox="0 0 18 18">
```
Comment (roopamjain01) [inline src/img/icons/add.svg:1]: Why did you change the viewbox for this and other buttons as well?
Could you please double check with the design?

File: src/js/components/Questionnaire/Summary.js [hunk 11]
```diff
@@ -294,23 +301,25 @@ class Summary extends Component<Props> {
...
+        </div>
         {submission.collaborators.length > 0 && (
-          <div className="collaborators-name-container">
+          <div>
             {submission.collaborators.map(({label}, index) =>
               {
-                return(<div key={index}>{label}</div>)
+                return(<div className="label" key={index}>{label}</div>)
```
Comment (roopamjain01) [inline src/js/components/Questionnaire/Summary.js:314]: Please use the different name for the classname, which are more related to the context.
Also font weight is 500 in figma but I can see it is 700 in the website.

White backgroun for component and collaborator are look slimmer in figma compare to the website. Please check for the status as well.

File: src/js/components/Questionnaire/Summary.js [hunk 12]
```diff
@@ -319,30 +328,95 @@ class Summary extends Component<Props> {
...
     return (
-      <div>
-        <h4>Request Information</h4>
-        <div><b>Product Name:</b> {submission.productName} </div>
-        <div><b>Submitted by:</b> {submitter.name}</div>
-        <div><b>Email:</b> {submitter.email}</div>
-        <div><b>Status:</b> {prettifyStatus(submission.status)}</div>
+      <div class="submission-details-container">
+        <h3>Submission details</h3>
+        <div class="row">
```
Comment (roopamjain01) [inline src/js/components/Questionnaire/Summary.js:361]: please use the more meaning name for the css class.

File: src/js/components/Questionnaire/Summary.js [hunk 13]
```diff
@@ -319,30 +328,95 @@ class Summary extends Component<Props> {
...
+          <div class="column">
```
Comment (roopamjain01) [inline src/js/components/Questionnaire/Summary.js:362]: same here.

File: src/js/components/Questionnaire/Summary.js [hunk 14]
```diff
@@ -319,30 +328,95 @@ class Summary extends Component<Props> {
...
+            <div className="product-name">{submission.productName}</div>
```
Comment (roopamjain01) [inline src/js/components/Questionnaire/Summary.js:363]: Have you confirmed what should display here?

Just because product name is the output of product type question and field.

File: src/js/components/Questionnaire/Summary.js [hunk 15]
```diff
@@ -292,25 +301,86 @@ class Summary extends Component<Props> {
...
+            <span className="submission-status"><img src={statusIcon}/>{status}</span>
+          </div>
+        </div>
+        <div class="submission-dates-submitter-info-row">
+          <div class="submission-details-column">
+            <span><b>Submission created: </b>{moment(submission.created).format("DD/MM/YYYY")}</span>
+            <span><b>Go live date: </b>{submission.releaseDate ? moment(submission.releaseDate).format("DD/MM/YYYY") : ''}</span>
+          </div>
+          <div className="submission-details-column">
+            <span className="submitter-name"><b>Submitted by: </b>{submitter.name}</span>
```
Comment (roopamjain01) [inline src/js/components/Questionnaire/Summary.js:355]: Span should close after "Submitted by:" and new span start for {submitter.name}.

File: src/js/components/Questionnaire/Summary.js [hunk 16]
```diff
@@ -292,25 +301,86 @@ class Summary extends Component<Props> {
...
+            {submission.questionnaireTitle}
+          </div>
+          <div class="submission-details-column">
+            <span className="submission-status"><img src={statusIcon}/>{status}</span>
+          </div>
+        </div>
+        <div class="submission-dates-submitter-info-row">
+          <div class="submission-details-column">
+            <span><b>Submission created: </b>{moment(submission.created).format("DD/MM/YYYY")}</span>
+            <span><b>Go live date: </b>{submission.releaseDate ? moment(submission.releaseDate).format("DD/MM/YYYY") : ''}</span>
```
Comment (roopamjain01) [inline src/js/components/Questionnaire/Summary.js:352]: same here as well. span tage should close after </b> tag. Please do the same for other as well.

File: src/js/components/Questionnaire/Summary.js [hunk 17]
```diff
@@ -292,25 +301,86 @@ class Summary extends Component<Props> {
...
+          </div>
+          <div className="submission-details-column">
+            <span className="submitter-name"><b>Submitted by: </b>{submitter.name}</span>
+            <span><b>Email: </b>{submitter.email}</span>
```
Comment (roopamjain01) [inline src/js/components/Questionnaire/Summary.js:356]: same here.

File: src/js/components/Questionnaire/Summary.js [hunk 18]
```diff
@@ -292,25 +301,86 @@ class Summary extends Component<Props> {
...
+
   renderCollboratorsInfo(submission: Submission, members) {
     return (
-      <div>
+      <div className="collaborators-container">
         <h4>Collaborators</h4>
-        <div><b>Add people to help complete your submission.</b></div>
+        <div className="collaborator-message">
+          You can add people to help complete your submission.
+          Please contact the&nbsp;<a href ={"mailto:" + this.props.securityTeamEmail}>security team</a>&nbsp;for more information.
```
Comment (roopamjain01) [inline src/js/components/Questionnaire/Summary.js:369]: please fix the underline issue for back button as well.

File: src/js/types/Questionnaire.js [hunk 19]
```diff
@@ -3,7 +3,7 @@
 // Questionnaire Schema
 
 import type {User} from "./User";
-import type {TaskSubmissionDisplay} from "./Task";
+import type {TaskSubmissionDisplay, ProductAspect} from "./Task";
```
Comment (roopamjain01) [inline src/js/types/Questionnaire.js:6]: where did you define "Product Aspect" type?

File: src/js/types/Questionnaire.js [hunk 20]
```diff
@@ -70,7 +70,10 @@ export type Submission = {
   hideWeightsAndScore: boolean,
   productName: string,
   riskResults: Array<RiskResult>,
-  collaborators: Array<User>
+  collaborators: Array<User>,
+  created: string,
+  releaseDate: string,
+  productAspects: Array<ProductAspect>
```
Comment (roopamjain01) [inline src/js/types/Questionnaire.js:76]: I can't see productAspect type. Please use productAspects: Array<*>

File: src/js/types/Questionnaire.js (hunk 19 above)
Comment (roopamjain01) [reply src/js/types/Questionnaire.js:6]: oh...I can see in the task file, but I guess it is wrong type. As far as I know product aspect is array of name(string).
